backend/users/views/user.py

```diff
@@ -47,35 +47,77 @@ def login(self, request, *args, **kwargs):
         except TokenError as e:
             raise InvalidToken(e.args[0])
 
-        token = serializer.validated_data
+        data = serializer.validated_data
 
-        expires = token.current_time + token.lifetime
+        refresh_token = data
+        access_token = data.access_token
 
-        tokenString = str(token)
-        headerPayload = '.'.join(tokenString.split('.')[0:2])
-        signature = tokenString.split('.')[2]
+        access_expires = access_token.current_time + access_token.lifetime
+        refresh_expires = refresh_token.current_time + refresh_token.lifetime
 
-        response = Response(token.payload, status=status.HTTP_200_OK)
+        accessTokenString = str(access_token)
+        headerPayload = '.'.join(accessTokenString.split('.')[0:2])
+        signature = accessTokenString.split('.')[2]
+
+        response = Response(access_token.payload, status=status.HTTP_200_OK)
+
+        host = self.get_host(request)
+
+        secure = not settings.DEBUG
+
+        response.set_cookie('header.payload', value=headerPayload, expires=access_expires, secure=secure, domain=host)
+        response.set_cookie('signature', value=signature, httponly=True, secure=secure, domain=host)
+        response.set_cookie('refresh', value=str(refresh_token), expires=refresh_expires, httponly=True, secure=secure,
+                            domain=host)
+
+        return response
+
+    @action(methods=['post'], detail=False)
+    def refresh(self, request, *args, **kwargs):
+        serializer = CustomTokenRefreshSerializer(data=request.data, context=self.get_serializer_context())
```

Comment:
Yes we could, but it will imply extracting the Validation on cookies, which is the work of the `validate` method of the serializer.
Although we do not want to return a `string` representation of the data as `TokenRefreshSerializer` `validate` method does:
```python
              data['refresh'] = str(refresh)

        return data
```